dpgen/generator/run.py

```diff
@@ -4437,16 +4437,19 @@ def post_fp_cp2k(iter_index, jdata, rfailed=None):
         sys_output = glob.glob(os.path.join(work_path, "task.%s.*/output" % ss))
         sys_output.sort()
         tcount += len(sys_output)
-        all_sys = None
+        log_file_path = os.path.join(work_path, f"{ss}.fp-fail.log")
+        all_sys = dpdata.MultiSystems(type_map=jdata["type_map"])
         for oo in sys_output:
-            _sys = dpdata.LabeledSystem(oo, fmt="cp2k/output")
-            # _sys.check_type_map(type_map = jdata['type_map'])
-            if all_sys is None:
-                all_sys = _sys
-            else:
-                all_sys.append(_sys)
+            with open(oo, 'r') as file:
+                content = file.read()
+            if 'SCF run NOT converged' in content:
```

Comment:
Yes, it will. But when using MultiSystems, the empty object will be added into that, for example create an empty "C0H0O0". This will raise error when save the MultiSystems to `data.xxx`. Maybe we should change the subroutine which you mentioned to skip instead of creating empty object.